### PR TITLE
feat: Cut off request/response body if its size exceeds limit #612

### PR DESCRIPTION
**Pull request by astsiapanay**

### CODE REVIEW

File: server/src/main/java/com/epam/aidial/core/server/log/GfLogStore.java [hunk 1]
```diff
@@ -189,10 +192,22 @@ private void append(ProxyContext context, LogEntry entry, String assembledStream
...
     private static void append(LogEntry entry, Buffer buffer) {
-        if (buffer != null) {
-            byte[] bytes = buffer.getBytes();
-            String chars = new String(bytes, StandardCharsets.UTF_8); // not efficient, but ok for now
-            append(entry, chars, true);
+        if (buffer == null) {
+            return;
+        }
+        boolean largeBuffer;
+        if (exceedLimit(buffer)) {
```
Comment (artsiomkorzun) [inline server/src/main/java/com/epam/aidial/core/server/log/GfLogStore.java]: simplify with  largeBuffer = exceedLimit(buffer)

File: server/src/main/java/com/epam/aidial/core/server/log/GfLogStore.java [hunk 2]
```diff
@@ -189,10 +192,22 @@ private void append(ProxyContext context, LogEntry entry, String assembledStream
...
+            buffer = buffer.slice(0, MAX_BODY_SIZE_BYTES);
```
Comment (artsiomkorzun) [inline server/src/main/java/com/epam/aidial/core/server/log/GfLogStore.java:204]: You can break utf-8 encoded tail. One character can be encoded as 4 bytes in utf-8.

Comment (artsiomkorzun) [reply server/src/main/java/com/epam/aidial/core/server/log/GfLogStore.java:204]: String constructor uses decoder that fixes this broken tail. Could you write a test to see what will be in this case?

File: server/src/main/java/com/epam/aidial/core/server/log/GfLogStore.java (hunk 1 above)
Comment (astsiapanay) [reply server/src/main/java/com/epam/aidial/core/server/log/GfLogStore.java]: fixed

File: server/src/main/java/com/epam/aidial/core/server/log/GfLogStore.java (hunk 2 above)
Comment (astsiapanay) [reply server/src/main/java/com/epam/aidial/core/server/log/GfLogStore.java:204]: good catch!

Comment (artsiomkorzun) [reply server/src/main/java/com/epam/aidial/core/server/log/GfLogStore.java:204]: The decoder will replace with character (65533). So it won't drop an exception at least.

Comment (astsiapanay) [reply server/src/main/java/com/epam/aidial/core/server/log/GfLogStore.java:204]: The decoder replaces broken bytes with 65533 character. 

It won't throw an exception.

**Review by artsiomkorzun — APPROVED** [review]

**Pull request merged**